src/test/java/com/iexec/core/replicate/ReplicateServiceTests.java

```diff
@@ -659,6 +660,32 @@ public void shouldNotUpdateReplicateStatusSinceAlreadyReported() {
                 .isEqualTo(Optional.empty());
     }
 
+    @Test
+    void shouldNotEncounterRaceConditionOnReplicateUpdate() {
+        Replicate replicate = new Replicate(WALLET_WORKER_1, CHAIN_TASK_ID);
+        replicate.updateStatus(REVEALING, ReplicateStatusModifier.WORKER);
+        ReplicatesList replicatesList = new ReplicatesList(CHAIN_TASK_ID, Collections.singletonList(replicate));
+
+        when(replicatesRepository.findByChainTaskId(CHAIN_TASK_ID)).thenReturn(Optional.of(replicatesList));
+        when(web3jService.isBlockAvailable(anyLong())).thenReturn(true);
+        when(iexecHubService.repeatIsRevealedTrue(anyString(), anyString())).thenReturn(true);
+        when(replicatesRepository.save(replicatesList)).thenReturn(replicatesList);
+
+        ReplicateStatusUpdate statusUpdate = ReplicateStatusUpdate.builder()
+                .modifier(WORKER)
+                .status(REVEALED)
+                .build();
+
+        // Without any synchronization mechanism,
+        // this would update 10 times to `REVEALED`.
```

Comment:
I think it would be interesting to run the same thing (10 updates) on the non-thread-safe method and assert that the wrong updates happened in because of concurrent threads (update to REVEALED more than once). That would highlight the efficiency of the new mechanism.

Reply:
Good idea, thanks! See https://github.com/iExecBlockchainComputing/iexec-core/pull/492/commits/949f44a332fbf89f6ef13998726f1d302096c581.